server.js

```diff
@@ -1,34 +1,93 @@
 import express from 'express'
-import bodyParser from 'body-parser'
+//import bodyParser from 'body-parser'
 import cors from 'cors'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+
+ import booksData from './data/books.json'
+
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('hallo world');
 })
 
+//Endpoint example: /books, /books/id/1, /books/book_pages_below_500/499, /books/book_pages_above_500/501
+//query example:    books?author=Rowling, books?title=harry, books?language=eng,  
+
+
+// get all the books and/or author/title/language with query parameters
+app.get('/books', (req, res) => {
+  const { author, title, language } = req.query
+
+  let booksToSend = booksData
+
+  if(author) {
+    booksToSend = booksToSend
+      .filter(book => book.authors.toLowerCase().includes(author.toLowerCase()))
+  }
+
+  if(title) {
+    booksToSend = booksToSend
+     .filter(book => book.title.toString().toLowerCase().includes(title.toLowerCase()))
+  }
+
+  if(language) {
+    booksToSend = booksToSend
+    .filter((book) => book.language_code === language)
+  }
+
+  res.json({length: booksToSend.length, data: booksToSend})
+});
+
+
+//get thin books (below 500 pages)
+app.get('/books/book_pages_below_500/:pages', (req, res) => {
+  const  pages = req.params.thin_books
+  const thinBooks = booksData.filter((book) => book.num_pages <= 500)
+
+  if(thinBooks) {
+    res.json({ length: thinBooks.length, data: thinBooks }) 
+  } else {
+    res.status(404).json(`Sorry, there is no thin books with ${pages} pages`)
+  }
+}) 
+
+
+//get thick books (above 500 pages)
+app.get('/books/book_pages_above_500/:pages', (req, res) => {
+  const  pages = req.params.thin_books
+  const thickBooks = booksData.filter((book) => book.num_pages >= 501)
+
+  if(thickBooks) {
+    res.json({ length: thickBooks.length, data: thickBooks }) 
+  } else {
+    res.status(404).json(`Sorry, there is no thick books with ${pages} pages`)
+  }
+}) 
```

Comment:
the two functions differ only in sign, so it would certainly be possible to create one function that would handle both cases